src/include/crc32c.h

```diff
@@ -26,7 +40,13 @@ extern ceph_crc32c_func_t ceph_choose_crc32(void);
  */
 static inline uint32_t ceph_crc32c(uint32_t crc, unsigned char const *data, unsigned length)
 {
+  if (!data && length > 16)
```

Comment:
@aclamk I proposed it multiple times, and it boils down to measuring average time needed for each implementation and for each block size and picking the point where one impl outperform other.